src/Symfony/Component/String/Utf8String.php

```diff
@@ -0,0 +1,249 @@
+        }
+
+        return $str;
+    }
+
+    public function replace(string $from, string $to): AbstractString
+    {
+        $str = clone $this;
+
+        if ('' === $from || !preg_match('//u', $from)) {
+            return $str;
+        }
+
+        if ('' !== $to && !preg_match('//u', $to)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        if ($this->ignoreCase) {
+            $str->string = implode($to, preg_split('{'.preg_quote($from).'}iu', $this->string));
+        } else {
+            $str->string = str_replace($from, $to, $this->string);
+        }
+
+        return $str;
+    }
+
+    public function slice(int $start = 0, int $length = null): AbstractString
+    {
+        $str = clone $this;
+        $str->string = (string) mb_substr($this->string, $start, $length, 'UTF-8');
+
+        return $str;
+    }
+
+    public function splice(string $replacement, int $start = 0, int $length = null): AbstractString
+    {
+        if (!preg_match('//u', $replacement)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $str = clone $this;
+        $start = $start ? \strlen(mb_substr($this->string, 0, $start, 'UTF-8')) : 0;
+        $length = $length ? \strlen(mb_substr($this->string, $start, $length, 'UTF-8')) : $length;
+        $str->string = substr_replace($this->string, $replacement, $start, $length ?? \PHP_INT_MAX);
+
+        return $str;
+    }
+
+    public function split(string $delimiter, int $limit = null, int $flags = null): array
+    {
+        if (1 > $limit = $limit ?? \PHP_INT_MAX) {
+            throw new InvalidArgumentException('Split limit must a positive integer.');
+        }
+
+        if ('' === $delimiter) {
+            throw new InvalidArgumentException('Split delimiter is empty.');
+        }
+
+        if (null !== $flags) {
+            return parent::split($delimiter.'u', $limit, $flags);
+        }
+
+        if (!preg_match('//u', $delimiter)) {
+            throw new InvalidArgumentException('Split delimiter is invalid UTF-8 string.');
+        }
+
+        $str = clone $this;
+        $chunks = $this->ignoreCase
+            ? preg_split('{'.preg_quote($delimiter).'}iu', $this->string, $limit)
+            : explode($delimiter, $this->string, $limit);
+
+        foreach ($chunks as &$chunk) {
+            $str->string = $chunk;
+            $chunk = clone $str;
+        }
+
+        return $chunks;
+    }
+
+    public function startsWith($prefix): bool
+    {
+        if ($prefix instanceof AbstractString) {
+            $prefix = $prefix->string;
+        } elseif (!\is_string($prefix)) {
+            return parent::startsWith($prefix);
+        }
+
+        if ('' === $prefix || !preg_match('//u', $prefix)) {
+            return false;
+        }
+
+        return 0 === ($this->ignoreCase ? mb_stripos($this->string, $prefix, 0, 'UTF-8') : strpos($this->string, $prefix));
```

Comment:
`strpos` → it should be `mb_strpos`?

Reply:
same answer as in https://github.com/symfony/symfony/pull/33553#discussion_r325831792